x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/expression/function/FunctionRegistryTests.java

```diff
@@ -166,18 +164,6 @@ public void testFunctionResolving() {
         assertThat(e.getMessage(), is("Cannot find function dummyfunction; this should have been caught during analysis"));
     }
 
-    public void testConfigurationOptionalFunction() {
-        UnresolvedFunction ur = uf(DEFAULT, mock(Expression.class));
-        FunctionRegistry r = new FunctionRegistry(
-            def(DummyConfigurationOptionalArgumentFunction.class, (Source l, Expression e, Configuration c) -> {
-                assertSame(e, ur.children().get(0));
-                return new DummyConfigurationOptionalArgumentFunction(l, List.of(ur), c);
-            }, "dummy")
-        );
-        FunctionDefinition def = r.resolveFunction(r.resolveAlias("DUMMY"));
-        assertEquals(ur.source(), ur.buildResolved(randomConfiguration(), def).source());
-    }
-
```

Comment:
While a bit paranoid, maybe let's port this to `EsqlFunctionRegistryTests` rather than just forgetting about it; we don't use `ConfigurationFunction` anymore, but we do have (and use) `EsqlConfigurationFunction`.